### PR TITLE
Bug 1732585: Modify kibana-proxy to rely on ServiceAccount as oauthcl…

### PR DESCRIPTION
**Pull request by jcantrill**
…ient

This bug resolves https://bugzilla.redhat.com/show_bug.cgi?id=1732585 by:
* Modifying kibana-proxy to rely on ServiceAccount as an oauthclient so that we no longer need to keep a secret and and oauthclient in sync

### CODE REVIEW

File: pkg/k8shandler/visualization.go [hunk 1]
```diff
@@ -473,11 +380,11 @@ func newKibanaPodSpec(cluster *logging.ClusterLogging, kibanaName string, elasti
...
 		"--https-address=:3000",
 		"-provider=openshift",
-		"-client-id=kibana-proxy",
-		"-client-secret-file=/secret/oauth-secret",
+		"-client-id=system:serviceaccount:openshift-logging:kibana",
+		"-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token",
 		"-cookie-secret-file=/secret/session-secret",
 		"-upstream=http://localhost:5601",
-		"-scope=user:info user:check-access user:list-projects",
+		"-scope=user:info user:check-access",
```
Comment (richm) [inline pkg/k8shandler/visualization.go]: we don't need list-projects here?

Comment (jcantrill) [reply pkg/k8shandler/visualization.go]: I'm uncertain why this is here as the Kibana proxy never needed to list projects.  All of that particular logic is in the openshift-elasticsearch-plugin.

Comment (jcantrill) [reply pkg/k8shandler/visualization.go]: backtracking here as https://bugzilla.redhat.com/show_bug.cgi?id=1746695 was introduced.. not understood why our access script doesnt expose it

**Comment by jcantrill** [comment]
Any cherrypick also needs https://github.com/openshift/cluster-logging-operator/pull/241

**Pull request merged**